apps/client/src/app/services/import-activities.service.ts

```diff
@@ -52,28 +55,56 @@ export class ImportActivitiesService {
       });
     }
 
-    await this.importJson({ content: activities });
+    return await this.importJson({ content: activities, dryRun });
   }
 
-  public importJson({ content }: { content: CreateOrderDto[] }): Promise<void> {
+  public importJson({
+    content,
+    dryRun = false
+  }: {
+    content: CreateOrderDto[];
+    dryRun?: boolean;
+  }): Promise<Activity[]> {
     return new Promise((resolve, reject) => {
-      this.postImport({
-        activities: content
-      })
+      this.postImport(
+        {
+          activities: content
+        },
+        dryRun
+      )
         .pipe(
           catchError((error) => {
             reject(error);
             return EMPTY;
           })
         )
         .subscribe({
-          next: () => {
-            resolve();
+          next: (data) => {
+            resolve(data.activities);
           }
         });
     });
   }
 
+  public importSelectedActivities(
+    selectedActivities: Activity[]
+  ): Promise<Activity[]> {
+    const importData: CreateOrderDto[] = [];
+    for (const activity of selectedActivities) {
+      importData.push({
```

Comment:
Extract to `convertToCreateOrderDto(aActivity: Activity)`

Reply:
Done